### PR TITLE
Add the parents attribute to the platform rule.

### LINKED ISSUE #6218: Platform reuse rules
It would be nice to be able to reuse existing platforms for defining other platforms. For example, we're using remote execution properties to turn on docker features like enabling network usage, or root access, or use a specific machine type for particular targets (multipool). Sometimes, extra constraints are needed to make sure the modified platform gets selected for a rule that has exec_compatible_with it. 

I think extending or modifying single parent platform would be a nice syntactic sugar to have, e.g:

platform_from_parent(
   parent,
   constraint_values=... / extra_constraint_values=...
   remote_properties=... / extra_remote_properties=...
)

### PR DESCRIPTION
**Pull request by katre**
Fixes #6218.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/rules/platform/Platform.java [hunk 1]
```diff
@@ -42,6 +45,18 @@ public ConfiguredTarget create(RuleContext ruleContext)
...
     PlatformInfo.Builder platformBuilder = PlatformInfo.builder().setLabel(ruleContext.getLabel());
 
+    List<PlatformInfo> parentPlatforms =
+        Lists.newArrayList(
+            PlatformProviderUtils.platforms(
+                ruleContext.getPrerequisites(PlatformRule.PARENTS_PLATFORM_ATTR, Mode.DONT_CHECK)));
+
+    if (parentPlatforms.size() > 1) {
+      throw ruleContext.throwWithAttributeError(
+          PlatformRule.PARENTS_PLATFORM_ATTR, "parents attribute must have a single value");
```
Comment (gregestren) [inline src/main/java/com/google/devtools/build/lib/rules/platform/Platform.java]: Optional nit: replace "parents" with PlatformRule.PARENTS_PLATFORM_ATTR to further consolidate the source of truth?

File: src/main/java/com/google/devtools/build/lib/rules/platform/PlatformRule.java [hunk 2]
```diff
@@ -52,14 +54,24 @@ public RuleClass build(RuleClass.Builder builder, RuleDefinitionEnvironment env)
...
         /* <!-- #BLAZE_RULE(platform).ATTRIBUTE(remote_execution_properties) -->
         A string used to configure a remote execution platform. Actual builds make no attempt to
         interpret this, it is treated as opaque data that can be used by a specific SpawnRunner.
         <!-- #END_BLAZE_RULE.ATTRIBUTE --> */
         .add(attr(REMOTE_EXECUTION_PROPS_ATTR, Type.STRING))
 
+        /* <!-- #BLAZE_RULE(platform).ATTRIBUTE(parents) -->
+        The label of a <code>platform</code> target that this platform should inherit from. Although
+        the attribute takes a list, there should be no more than one platform present. Any
+        constraint_settings not set directly on this platform will be found in the parent platform.
```
Comment (gregestren) [inline src/main/java/com/google/devtools/build/lib/rules/platform/PlatformRule.java]: Did we have API reference to the precise semantics? (what happens if both platforms define the constraint_setting, {PARENT_REMOTE_EXECUTION_PROPERTIES}, etc)?

File: src/main/java/com/google/devtools/build/lib/rules/platform/Platform.java (hunk 1 above)
Comment (katre) [reply src/main/java/com/google/devtools/build/lib/rules/platform/Platform.java]: Done.

File: src/main/java/com/google/devtools/build/lib/rules/platform/PlatformRule.java (hunk 2 above)
Comment (katre) [reply src/main/java/com/google/devtools/build/lib/rules/platform/PlatformRule.java]: This is the API reference, I can add more detail if anything is unclear.

Comment (gregestren) [reply src/main/java/com/google/devtools/build/lib/rules/platform/PlatformRule.java]: For the sake of precision, can we clarify that the child value overrides the parent value if both are set (even if that's intuitive anyway)? Is it worth mentioning the use of {PARENT_REMOTE_EXECUTION_PROPERTIES} here?

This seems like the right place o be pedantically detailed.

Comment (gregestren) [reply src/main/java/com/google/devtools/build/lib/rules/platform/PlatformRule.java]: Looks great!

**Pull request closed without merging**